### PR TITLE
Add recipe for moving data to unified memory

### PR DESCRIPTION
**Pull request by davidbeckingsale**

### CODE REVIEW

**Comment by davidbeckingsale** [comment]
@Ram3sh - you asked about this a while back, now we have the "cookbook", here is the recipe!

File: docs/sphinx/cookbook/move_to_managed.rst [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+When using a system with NVIDIA GPUs, you may realize that some host data
+should be moved to unified memory in order to make it accessible by the GPU.
+You can do this with the :func:`umpire::ResourceManager::move` operation:
+
+.. literalinclude:: ../../../examples/cookbook/recipe_move_to_managed.cpp
+                    :lines: 34
+
+The move operation will copy the data from host memory to unified memory,
+allocated using the provided ``um_allocator``. The complete example is included
```
Comment (mcfadden8) [inline docs/sphinx/cookbook/move_to_managed.rst]: I believe that this operation will also deallocate the memory used from the source of the data move.  I noticed that this is stated in the comment of the cookbook source, but perhaps it also merits being stated here?

Comment (davidbeckingsale) [reply docs/sphinx/cookbook/move_to_managed.rst]: Good idea, I have addressed this

**Comment by davidbeckingsale** [comment]
@mcfadden8 - this should be ready

**Pull request merged**